src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/LambdaCompiler.Expressions.cs

```diff
@@ -796,15 +796,10 @@ private void EmitMemberAssignment(BinaryExpression node, CompilationFlags flags)
             }
             else
             {
-                var prop = member as PropertyInfo;
-                if ((object)prop != null)
-                {
-                    EmitCall(objectType, prop.GetSetMethod(true));
-                }
-                else
-                {
-                    throw Error.InvalidMemberType(member);
-                }
```

Comment:
@VSadov, are you ok with this change?  It looks safe, but it also looks like the author of this code was future proofing a bit.